src/csm_testing/tests/sat_functional/firmware/test_firmware.py

```diff
@@ -0,0 +1,188 @@
+#
+# Permission is hereby granted, free of charge, to any person obtaining a
+# copy of this software and associated documentation files (the "Software"),
+# to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense,
+# and/or sell copies of the Software, and to permit persons to whom the
+# Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included
+# in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+#
+import unittest
+import subprocess
+import shlex
+import re
+import os
+
+SAT_FIRMWARE_HEADER = ['xname', 'name', 'target_name', 'version']
+
+def get_xnames_from_firmware_output() -> list:
+    """Run 'sat firmware' and return a list of unique xnames from the output."""
+    command = "sat firmware"
+    proc = subprocess.run(shlex.split(command), stdout=subprocess.PIPE, stderr=subprocess.PIPE, check=True)
+    output = proc.stdout.decode().strip()
+    lines = output.splitlines()
+    xnames = []
+    for line in lines[2:]:
+        fields = [col.strip() for col in line.split('|') if col.strip()]
+        if fields and fields[0].startswith('x') and fields[0] not in xnames:
+            xnames.append(fields[0])
+    return xnames
+
+class TestFirmware(unittest.TestCase):
+
+    def test_firmware_creates_and_deletes_snapshot(self):
+        """Test that running `sat firmware` creates a snapshot and outputs a report"""
+        command = "sat firmware"
+        proc = subprocess.run(shlex.split(command), stdout=subprocess.PIPE, stderr=subprocess.PIPE, check=True)
+        output = proc.stdout.decode().strip()
+        info_msg = proc.stderr.decode().strip()
+
+        # Extract the snapshot name from the info message
+        pattern = (
+            r'INFO: Snapshot (\S+) created successfully\.\n'
+            r'INFO: Use `sat firmware --delete-snapshot` to delete the snapshot if it is no longer needed'
+        )
+        match = re.search(pattern, info_msg)
+        self.assertIsNotNone(match, "Snapshot creation message not found in info_msg.")
+        snapshot_name = match.group(1)
+
+        # Check that the output contains the expected table header
+        for col in SAT_FIRMWARE_HEADER:
+            self.assertIn(col, output)
+
+        # Now delete the snapshot to keep the system clean
+        delete_command = f"sat firmware --delete-snapshot {snapshot_name}"
```

Comment:
This is good to test in a test case like this, but we also want to make sure all snapshots from other tests are cleaned up. To do so, add a `tearDown` method. That `tearDown` method could either:

* find and delete all SAT snapshots that have the expected prefix (perhaps between the start and end time of the tests executed in the class).
* or delete all SAT snapshots that have been explicitly marked for deletion. To do this, you'd need to mark them for deletion after they're created. That's probably a less robust solution.